### PR TITLE
fix(swaps): add resolve request validation slippage

### PR DESCRIPTION
**Pull request by ghost**
In this commit we're relaxing raiden's resolve request validation by 3 blocks in order to prevent an issue where new blocks are mined during the payment.

This is a quick fix to the issue described in: https://github.com/ExchangeUnion/xud/issues/1184#issuecomment-529089822

As a follow-up I'd like to add a simulation test scenario to make sure maker still has sufficient buffer.

### CODE REVIEW

File: lib/swaps/Swaps.ts [hunk 1]
```diff
@@ -739,7 +739,8 @@ class Swaps extends EventEmitter {
         source = 'Taker';
         destination = 'Maker';
         const lockExpirationDelta = expiration - chain_height;
-        if (deal.makerCltvDelta! > lockExpirationDelta) {
+        const LOCK_EXPIRATION_SLIPPAGE = 3;
+        if (deal.makerCltvDelta! - LOCK_EXPIRATION_SLIPPAGE > lockExpirationDelta) {
```
Comment (offerm) [inline lib/swaps/Swaps.ts]: consider to add the LOCK_EXPIRATION_SLIPPAGE to the maker's final_cltv_delta  and here just check without the LOCK_EXPIRATION_SLIPPAGE.

Comment (erkarl) [reply lib/swaps/Swaps.ts]: I don't understand. In the case of raiden there's no concept of `final_cltv_delta`. The way I understand it is that we only want to add the slippage for raiden payments from taker to maker.

**Comment by offerm** [comment]
There can also be a slippage on the LN payment. Same reason.

The maker, when calculating the lock_duration is using the route to the
taker + ~24 hours. This is converted to the maker blocks. To this we should
add 3 blocks before we give to the taker.

When getting the payment we should make a normal check (as it is now,
without the 3 blocks)

**Comment by erkarl** [comment]
> There can also be a slippage on the LN payment. Same reason.

The `cltv_expiry` for LN payment is relative. How can there be slippage?

**Review by sangaman — APPROVED** [review]
Looks good to me, I'd only recommend a line or two of comments explaining what the slippage is and why we must account for it, so in the future it's not a mystery to someone reading the code that's unfamiliar with the concept.

As for applying this to the lnd invoices, I'm also under the impression that it's not necessary since the values are relative. Ultimately the check that we have sufficient cltv delay happens within lnd code as I understand it, not xud (we just pass lnd the relative value when adding an invoice).

**Comment by kilrau** [comment]
> Looks good to me, I'd only recommend a line or two of comments explaining what the slippage is and why we must account for it, so in the future it's not a mystery to someone reading the code that's unfamiliar with the concept.

Agree, these comments would be very nice :)

> As for applying this to the lnd invoices, I'm also under the impression that it's not necessary since the values are relative. Ultimately the check that we have sufficient cltv delay happens within lnd code as I understand it, not xud (we just pass lnd the relative value when adding an invoice).

That's also my understanding, could you confirm or explain what you mean so that this can be merged asap? @offerm

**Comment by kilrau** [comment]
I just tested this branch with a native simnet setup. 1. leg lightning, 2. leg raiden orders (`xucli buy 1 btc/dai 10000`) still not working. Error same: `09/09/2019 12:44:46.856 [SWAPS] debug: deal 92521de004cd68ff57f8854d895b25f62b6775bf101d70ba135de4d502bd57bc failed due to SwapTimedOut: undefined`

Isn't this supposed to fix this or am I missing something?

**Comment by kilrau** [comment]
Oh, the xud1.simnet.exchangeunion.com would need to be updated too I guess.

**Comment by kilrau** [comment]
There are some other weird entries with the node key being replaced by `undefined`. Do we know why is that?

**Comment by sangaman** [comment]
Those logs show `09/09/2019 10:44:18.393 [SWAPS] error: cltvDelta of 7359 does not meet 7360 minimum` which indicates that it's failing due to the bug that's fixed here, so we're all good on that front.

I'll look into that node key being replaced by undefined issue, definitely separate from this though.

**Review by sangaman — APPROVED** [review]

**Pull request merged**